src/term.rs

```diff
@@ -146,6 +149,12 @@ pub enum Term {
     ParseError,
 }
 
+impl Default for Term {
+    fn default() -> Self {
+        Term::Null
+    }
+}
```

Comment:
Why do we need this? It looks pretty dubious.

Reply:
In Rust, `Default` is useful to swap values in place without violating borrowing rules. The typical use-case is to take some value out replacing it by a default value, doing what you have do, and put it back in place:

```
fn map_sth(t: Term) -> Term { ... }

fn map_in_place(t_ref: &mut Term) {
  let t = std::mem::take(t_ref); // Take content and replace it by Default::default()
  *t_ref = map_sth(t);
}
```

This is one reason it is quite liberally implemented in Rust. Also I don't find `null` to be unreasonable, when you need a dummy term for testing or something. That being said, this one is most probably an artifact for a previous use with `mem::take`, as removing it doesn't fail compilation. I can remove it for now.

Reply:
I see.

The documentation is pretty tricky

> A trait for giving a type a useful default value.

I would interpret “useful” in this sentence to mean “meaningful”. But the next sentence is

> Sometimes, you want to fall back to some kind of default value, and don’t particularly care what it is.

Which contradicts my understanding of the first sentence.

Anyway, with the disclaimer that `default()` is really a placeholder value that you never want to inspect, I have no qualm. (though, of course, if it's dead code, maybe it's best to kill it).